### PR TITLE
reactivate AD tests: FiniteGP

### PR DESCRIPTION
**Pull request by st--**
Follow-up to #313

### CODE REVIEW

File: test/finite_gp_projection.jl [hunk 1]
```diff
@@ -177,42 +149,40 @@ end
...
+            atol=1e-8,
+            rtol=1e-8,
+        )
+
+        # Check that the gradient w.r.t. the noise is approximately correct for `f`.
+        σ_ = randn(rng)
+        adjoint_test((σ_, ŷ) -> logpdf(f(x, exp(σ_)), ŷ), l̄, σ_, ŷ)
+        adjoint_test((σ_, Ŷ) -> sum(logpdf(f(x, exp(σ_)), Ŷ)), l̄, σ_, Ŷ)
+
+        # Check that the gradient w.r.t. a scaling of the GP works.
```
Comment (st--) [inline test/finite_gp_projection.jl:180]: It's not actually implemented in AbstractGPs, so we should remove these (presumably leftover from the initial Stheno port?)

File: test/sparse_approximations.jl [hunk 2]
```diff
@@ -122,11 +122,47 @@ end
...
+        x = collect(range(-3.0, 3.0; length=N))
+        # @testset "dense" begin
+        #     rng = MersenneTwister(123456)
+        #     A = randn(rng, N, N - 2)
+        #     adjoint_test(
+        #         (x, A) -> begin
+        #             f = GP(sin, SqExponentialKernel())
+        #             Σy = _to_psd(A)
+        #             C = cholesky(Σy)
+        #             return AbstractGPs.tr_Cf_invΣy(f(x, Σy), Σy, C)
```
Comment (st--) [inline test/sparse_approximations.jl:145]: This isn't actually implemented. I'm wondering if we should just remove these tests entirely. Not sure what the value is in testing gradients of an internal function specifically - as long as the logpdf and mean/cov gradients work fine ... just remove it?

File: test/sparse_approximations.jl [hunk 3]
```diff
@@ -122,11 +122,47 @@ end
     σ² = 1.234
     Σy_Diagonal = Diagonal(Fill(σ², N))
     Σy_ScalMat = ScalMat(N, σ²)
+    Σy_dense = _to_psd(randn(N, N - 2))
     f = GP(SqExponentialKernel())
 
-    for Σy in (Σy_Diagonal, Σy_ScalMat)
+    for Σy in (Σy_Diagonal, Σy_ScalMat) #, Σy_dense)
```
Comment (st--) [inline test/sparse_approximations.jl:128]: it's not implemented for dense :see_no_evil:

File: test/finite_gp_projection.jl (hunk 1 above)
Comment (willtebbutt) [reply test/finite_gp_projection.jl:180]: Yup. These can go.

File: test/sparse_approximations.jl (hunk 3 above)
Comment (willtebbutt) [reply test/sparse_approximations.jl:128]: Probably for the best for now. If the observation covariance is dense, I would _imagine_ that performance will generally be worse than exact inference.

File: test/sparse_approximations.jl (hunk 2 above)
Comment (willtebbutt) [reply test/sparse_approximations.jl:145]: I'm also happy for these tests to be removed.

**Review by willtebbutt — APPROVED** [review]
I'm happy with this, subject to the comments.

**Pull request merged**